spring-data-jpa/src/test/java/org/springframework/data/jpa/domain/sample/ItemSiteId.java

```diff
@@ -40,14 +42,12 @@ public ItemSiteId(ItemId item, Integer site) {
 	public boolean equals(Object o) {
 		if (this == o)
 			return true;
-		if (!(o instanceof ItemSiteId))
+		if (!(o instanceof ItemSiteId that))
 			return false;
 
-		ItemSiteId that = (ItemSiteId) o;
-
-		if (item != null ? !item.equals(that.item) : that.item != null)
+		if (!Objects.equals(item, that.item))
 			return false;
-		return site != null ? site.equals(that.site) : that.site == null;
+		return Objects.equals(site, that.site);
 	}
```

Comment:
We have a new template for `equals` at https://github.com/spring-projects/spring-data-build/blob/main/etc/ide/IntelliJ.equals-template.txt, which I'll run this by, to ensure we are compliant.